### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by GregAC**

### CODE REVIEW

**Review by andreaskurth — APPROVED** [review]
LGTM, thanks for these fixes and improvements, @GregAC!

File: sw/common/spi.c [hunk 1]
```diff
@@ -6,21 +6,16 @@
 #include "spi.h"
 #include "dev_access.h"
 
-typedef struct spi_handle{
-    spi_reg_t *reg;
-    uint32_t speed;
-} spi_handle_t;
-
-void spi_init(spi_t *spi, spi_reg_t *spi_reg, uint32_t speed){
+void spi_init(spi_t *spi, spi_reg_t spi_reg, uint32_t speed){
```
Comment (marnovandermaas) [inline sw/common/spi.c:9]: Nit: space before {

**Review by marnovandermaas — APPROVED** [review]
LGTM

**Review by ctopal — APPROVED** [review]
LGTM

**Pull request merged**